apis/gvkmanifest/v1alpha1/gvkmanifest_types.go

```diff
@@ -0,0 +1,43 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GVKManifestSpec struct {
+	Groups []Group `json:"groups,omitempty"`
+}
+
+type Group struct {
```

Comment:
The problem is group name could be an empty string, meaning it can't be the key in a yaml dict I believe

Reply:
I can definitely do that, as long as we're ok with the empty group name case: "" is an allowed key, but may be confusing on first glance

Reply:
cc @maxsmythe

Reply:
Are you referencing the empty group name for things like Pod?

```
apiVersion: "v1",
kind: "Pod",
```

If you need to have something there, you could potentially put `core` as that is what that stuff is referred generally.

That said, I think you would be better off omitting the `Group` field when there is no proper group to reference since this is already a pattern in k8s (assuming that was what you were referencing).

Reply:
Yeah, that's the case I'm thinking of. Post-switching to the maps model, a GVK manifest with an empty group name looks like the following in yaml, which is technically valid yaml but does look a little funny:
`
apiVersion: gvkmanifest.gatekeeper.sh/v1alpha1
kind: GVKManifest
metadata:
  name: gvkmanifest
spec:
  groups:
    "":
      "v1": ["Service"]
    "apps"
      "v1": ["Deployment"]
`